### ISSUE TITLE
analogRead() takes about 2.4ms on a MZ ECG silicone running  @ 200MHz.

### ISSUE THREAD

**Issue by JacobChrist** [chipKIT32/chipKIT-core#317]
This is really really slow considering that MZ EFG silicone takes about 4.2us @ 200MHz

**Comment by EmbeddedMan** [comment]
Is this with your new non-blocking ADC read change?

**Comment by majenkotech** [comment]
The whole EC ADC is so broken it has had to be fudged around - and that makes it really slow.

Not alot that can be done about that. Just read the comments in ECADC.c...

    // must throw out first 16 samples
    // keep the 17th. We can not access any ADC registers
    // however we can look at the DMA to see when things complete

and

            // we have noticed problems that the DMA channel is not always
            // triggered, so after a time out value, just try again.
            // fundamentally the problem is that AD1FLTR6bits.AFEN must
            // be reset for each oversample conversion, disable and reenabled.

            // we know a conversion is going to take 8000ns * 16, or 128 us
            // we don't want to check too often so DMA and ADC can work
            // but we don't want to wait too long and hold things up

The presence of a number of loops and `delayMicroseconds()` is a big cause of the delay.

**Comment by JacobChrist** [comment]
No, see Matt's comments. Wasn't sure if anything could be done, but I wanted to capture the issue.

**Comment by majenkotech** [comment]
Oh, and don't get your silicon and silicone mixed up... silicon's what chips are made of - silicone's what you put in breasts.

**Comment by JacobChrist** [comment]
Apparently the MZ ECG is made of silicone!

**Comment by majenkotech** [comment]
Ah, that explains why it went tits up then ;)

**Comment by EmbeddedMan** [comment]
Oh, I knew this was coming. Being located in the LA area, I'm sure silicone
just comes up on Jacob's spellcheck much more often than silicon. :-)
